libcontainerd/client_linux.go

```diff
@@ -0,0 +1,98 @@
+package libcontainerd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/docker/docker/pkg/idtools"
+	"github.com/opencontainers/specs"
+)
+
+// CreateContainer defines methods for container creation.
+type CreateContainer interface {
```

Comment:
`CreateContainer` seems like a weird name for an interface. Perhaps `ContainerCreator`?